src/services/user-group/hooks/helpers/classHooks.js

```diff
@@ -39,4 +42,42 @@ const saveSuccessor = async (context) => {
 	return context;
 };
 
-module.exports = { prepareGradeLevelUnset, sortByGradeAndOrName, saveSuccessor };
+const restrictFINDToUsersOwnClasses = async (context) => {
+	const currentUser = await context.app
+		.service('users')
+		.get(context.params.account.userId, { query: { $populate: 'roles' } });
+	const skipHook = currentUser.roles.filter((u) => u.name === 'superhero' || u.name === 'administrator').length > 0;
+	if (skipHook) return context;
+
+	const { _id } = currentUser;
+	const restrictQuery = { $or: [{ userIds: _id }, { teacherIds: _id }] };
+	const originalQuery = context.params.query;
+	context.params.query = { $and: [restrictQuery, originalQuery] };
+
+	return context;
+};
+
+const restrictToUsersOwnClasses = async (context) => {
+	const currentUser = await context.app
+		.service('users')
+		.get(context.params.account.userId, { query: { $populate: 'roles' } });
+	const skipHook = currentUser.roles.filter((u) => u.name === 'superhero' || u.name === 'administrator').length > 0;
+	if (skipHook) return context;
+
+	const klass = await context.app.service('classModel').get(context.id);
+
+	const userId = context.params.account.userId.toString();
+	if (!(klass.userIds.some((u) => equalIds(u, userId)) || klass.teacherIds.some((u) => equalIds(u, userId)))) {
+		throw new NotFound('class not found');
```

Comment:
it should be Forbidden

Reply:
that would tell the attacker that the classid is valid, an information he could use in further attacks.
generally, when calling an object the attacker shouldnt know about, the behavior should be the same as if it wouldnt exist. Its the same for example when trying to access objects on other schools.